### ISSUE TITLE
CLI: Typings for main.ts

### ISSUE THREAD

**Issue by shilman** [storybookjs/storybook#11843]
Clean and consolidated typings for configuration files:
- [ ] Type definitions
- [ ] Add to CLI templates

```ts
import { MainConfig } from '@storybook/react';

export default {
  stories: ['../src/**/*.stories.*'],
  addons: ['@storybook/addon-essentials'],
} as Main;
```

**Comment by alvis** [comment]
With types already available from `@storybook/react/types`, maybe we can close this issue?

**Comment by shilman** [comment]
Addressed in 7.0 https://storybook.js.org/docs/react/api/main-config
